### ISSUE TITLE
terraform-vagrant :: Incompatible API version - error

### ISSUE THREAD

**Issue by rbgithuub** [bmatcuk/terraform-provider-vagrant#2]
I am trying to work with terraform and vagrant to spin-up my local development infrastructure.

I have my custom plugin provider for vagrant in windows environment under the path 

%AppData%\Roaming\terraform.d\plugins\windows_amd64\

Versions:
========
Oracle VirtualBox: VirtualBox-6.0.8-130520-Win
Terraform: 0.12.3
Vagrant: vagrant_2.2.4_x86_64
terraform-provider-vagrant: terraform-provider-vagrant_v1.1.1

When  I issue the command terraform init, the provider is identified properly as below:

> >terraform init

Initializing the backend...

Initializing provider plugins...

The following providers do not have any version constraints in configuration,
so the latest version was installed.

To prevent automatic upgrades to new major versions that may contain breaking
changes, it is recommended to add version = "..." constraints to the
corresponding provider blocks in configuration, with the constraint strings
suggested below.

* provider.vagrant: version = "~> 1.1"

Terraform has been successfully initialized!

You may now begin working with Terraform. Try running "terraform plan" to see
any changes that are required for your infrastructure. All Terraform commands
should now work.

If you ever set or change modules or backend configuration for Terraform,
rerun this command to reinitialize your working directory. If you forget, other
commands will detect it and remind you to do so if necessary.

Whereas when I issue the command terraform plan against my default terraform.tf file, I get below error:

> terraform plan

> Error: Failed to instantiate provider "vagrant" to obtain schema: Incompatible API version with plugin. Plugin version: 4, Client versions: [5]

Please suggest if I should downgrade my version of vagrant  software to be compatible with the terraform-provider-vagrant custom plugin.?

**Comment by unclesamwk** [comment]
+1

**Comment by bmatcuk** [comment]
Hi @rbgithuub and @unclesamwk. I'm really, really sorry I didn't see this issue until just now. For some reason, I didn't get any notification that this issue was created. Not sure what happened! =( This is related to #1: I built this provider for terraform v0.11 which appears to be incompatible with v0.12. It's been a while since I've worked on this project, but I do plan on upgrading it to support v0.12, but it might be a while. In the meantime, you'll have to downgrade to terraform v0.11 unfortunately.
